internal/kittehs/hash.go

```diff
@@ -0,0 +1,43 @@
+package kittehs
+
+import (
+	"bytes"
+	"cmp"
+	"crypto/sha256"
+	"encoding/gob"
+	"encoding/hex"
+	"fmt"
+	"slices"
+)
+
+// Hash computes the SHA256 hash of the given input.
```

Comment:
Since we're hashing files or tar archives that are already a sequence of bytes, I don't see the value of this.

Reply:
it is a helper to make hashing of maps deterministic.